### PR TITLE
UCP/RNDV/GTEST: Fix RNDV GET_Zcopy initialization of user's memh

### PR DESCRIPTION
**Pull request by dmitrygx**
## What

Fix RNDV GET_Zcopy initialization of user's memh.

## Why ?

`req->send.rndv.lanes_map_all` contains indices of array RNDV GET Zcopy lanes.

## How ?

Go through `rndv_req->send.rndv.lanes_map_all` and lookup lane from `ucp_ep_config(ep)->rndv.get_zcopy.lanes` array.

### CODE REVIEW

**Comment by dmitrygx** [comment]
@hoopoepg @yosefe could you review pls?

**Review by hoopoepg — DISMISSED** [review]

**Comment by yosefe** [comment]
@dmitrygx any way to simplify the test?

**Comment by dmitrygx** [comment]
> @dmitrygx any way to simplify the test?

@yosefe the major point which increases complexity of the test is saving/resetting/restoring of list of `ucp_context_h`'s MDs.
I though about checking memh used in a RNDV request, but we don't have an access to a RNDV request from a receive request returned from `ucp_am_recv_data_nbx()`

File: test/gtest/ucp/test_ucp_sockaddr.cc [hunk 1]
```diff
@@ -2432,17 +2439,34 @@ class test_ucp_sockaddr_protocols : public test_ucp_sockaddr {
...
+        for (int i = 0; i < num_iters; i++) {
             std::string sb(size, 'x');
             std::string rb(size, 'y');
             std::string shdr(hdr_size, 'x');
             std::string rhdr(hdr_size, 'y');
+            ucp_mem_h rmemh(NULL);
+
+            if (recv_prereg_memh) {
+                rmemh = receiver().mem_map(&rb[0], size);
+                if (recv_mds != NULL) {
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_sockaddr.cc:2458]: 1. why here and not in the test itself?
2. can this be outside of `if (recv_prereg_memh) {` ?

File: test/gtest/ucp/test_ucp_sockaddr.cc [hunk 2]
```diff
@@ -2460,6 +2484,19 @@ class test_ucp_sockaddr_protocols : public test_ucp_sockaddr {
             compare_buffers(shdr, rhdr);
 
             set_am_data_handler(receiver(), 0, NULL, NULL);
+
+            if (recv_prereg_memh) {
+                if (recv_mds != NULL) {
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_sockaddr.cc:2489]: can this be outside of if (recv_prereg_memh) { ?

File: test/gtest/ucp/test_ucp_sockaddr.cc [hunk 3]
```diff
@@ -2658,6 +2695,18 @@ UCS_TEST_P(test_ucp_sockaddr_protocols, am_rndv_64k, "RNDV_THRESH=0")
     test_am_send_recv(64 * UCS_KBYTE);
 }
 
+UCS_TEST_P(test_ucp_sockaddr_protocols,
+           am_rndv_64k_recv_prereg_single_rndv_get_zcopy_lane, "RNDV_THRESH=0",
+           "MAX_RNDV_LANES=1", "RNDV_SCHEME=get_zcopy")
+{
+    /* The test checks that memory registration won't happen during RNDV GET
+     * Zcopy if a memory buffer was preregistered */
+    uct_md_h *recv_mds = (uct_md_h*)alloca(sizeof(*recv_mds) *
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_sockaddr.cc:2704]: IMO std::vector will be nicer

Comment (dmitrygx) [reply test/gtest/ucp/test_ucp_sockaddr.cc:2704]: done

**Review by yosefe — APPROVED** [review]

**Comment by dmitrygx** [comment]
@yosefe could you re-approve pls?

**Review by yosefe — APPROVED** [review]

**Pull request merged**